docs/sql-ref-syntax-qry-select-transform.md

```diff
@@ -0,0 +1,242 @@
+        [ LINES TERMINATED BY row_terminated_char ]
+        [ NULL DEFINED AS null_char ]
+```
+
+### Parameters
+
+* **expression**
+    
+    Specifies a combination of one or more values, operators and SQL functions that results in a value.
+
+* **row_format**    
+
+    Spark uses the `SERDE` clause to specify a custom SerDe for one table. Otherwise, use the `DELIMITED` clause to use the native SerDe and specify the delimiter, escape character, null character and so on.
+
+* **SERDE**
+
+    Specifies a custom SerDe for one table.
+
+* **serde_class**
+
+    Specifies a fully-qualified class name of a custom SerDe.
+
+* **DELIMITED**
+
+    The `DELIMITED` clause can be used to specify the native SerDe and state the delimiter, escape character, null character and so on.
+
+* **FIELDS TERMINATED BY**
+
+    Used to define a column separator.
+    
+* **COLLECTION ITEMS TERMINATED BY**
+
+    Used to define a collection item separator.
+   
+* **MAP KEYS TERMINATED BY**
+
+    Used to define a map key separator.
+    
+* **LINES TERMINATED BY**
+
+    Used to define a row separator.
+    
+* **NULL DEFINED AS**
+
+    Used to define the specific value for NULL.
+    
+* **ESCAPED BY**
+
+    Used for escape mechanism.
+
+* **RECORDWRITER**
+
+    Specifies a fully-qualified class name of a custom RecordWriter. The default value is `org.apache.hadoop.hive.ql.exec.TextRecordWriter`.
+
+* **RECORDREADER**
+
+    Specifies a fully-qualified class name of a custom RecordReader. The default value is `org.apache.hadoop.hive.ql.exec.TextRecordReader`.
+
+* **command_or_script**
+
+    Specifies a command or a path to script to process data.
+
+### SerDe behavior
+
+Spark uses Hive Serde `org.apache.hadoop.hive.serde2.lazy.LazySimpleSerDe` by default, columns will be casted
+to `STRING` and combined by tabs before feeding to the user script. All `NULL` values will be converted
+to the literal string `"\N"` in order to differentiate `NULL` values from empty strings. The standard output of the
+user script will be treated as TAB-separated STRING columns, any cell containing only `"\N"` will be re-interpreted
+as a `NULL`, and then the resulting STRING column will be cast to the data type specified in the table declaration
+in the usual way. If the actual number of output columns is less than the number of specified output columns,
+insufficient output columns will be supplemented with `NULL`. If the actual number of output columns is more than the
+number of specified output columns, the output columns will only select the corresponding columns and the remaining part
+will be discarded. If there is no `AS` clause after `USING my_script`, Spark assumes that the output of the script contains 2 parts:
+
+   1. key: which is before the first tab.
+   2. value: which is the rest after the first tab.
+
+If there is no enough tab, Spark will return `NULL` value in `SERDE` mode or throw `ArrayOutOfBoundsException` in `DELIMITED` mode.
```

Comment:
does hive also throw `ArrayOutOfBoundsException`?

Reply:
> does hive also throw `ArrayOutOfBoundsException`?

No, will pad `NULL` value

Reply:
Can we fix this bug in Spark first?

Reply:
Have been fixed in https://github.com/apache/spark/pull/29421/files

Reply:
Why do we still document `ArrayOutOfBoundsException` if it has already been fixed?

Reply:
> Why do we still document `ArrayOutOfBoundsException` if it has already been fixed?

Since this is not fixed in spark-3.0 and this doc work for 3.0/3.1/master. So plan to update this in next follow up pr.
So I need to update this desc like not support alias too?

Reply:
Can we fix it in 3.0/3.1 as well? I feel it's weird to decribe a bug in the document.

Reply:
> Can we fix it in 3.0/3.1 as well? I feel it's weird to describe a bug in the document.

Yea，rais  a pr later

Reply:
> Can we fix it in 3.0/3.1 as well? I feel it's weird to describe a bug in the document.

This doc updated